### PR TITLE
make 'summarize_dt' work with multiple value columns

### PR DESCRIPTION
**Pull request by chacalle**
## Describe changes

Make `summarize_dt` work with multiple value columns so that the user doesn't have to manually call the function multiple times.

This has a breaking change, since the argument `value_col` is changed to `value_cols`. Here are all the places that would need to be updated https://github.com/search?p=2&q=org%3Aihmeuw-demographics+summarize_dt&type=Code. It looks like popMethods and anything that used `model_template`.

Maybe can introduce this change in the next docker builds?

### Packages Repositories

* [X] Have you read the [contributing guidelines](https://github.com/ihmeuw-demographics/packageTemplate/wiki#guide-to-r-package-development) for `ihmeuw-demographics` R packages?
* [X] Have you successfully run `devtools::check()` locally?
* [X] Have you updated or added function (and vignette if applicable) documentation? Did you update the 'man' and 'NAMESPACE' files with `devtools::document()`?
* [X] Have you added in tests for the changes included in the PR?
* [X] Do the changes follow the `ihmeuw-demographics` [code style](https://github.com/ihmeuw-demographics/packageTemplate/wiki/Code-style-guide)?
* [ ] Do the changes need to be immediately included in a new build of [`docker-base`](https://github.com/ihmeuw-demographics/docker-base) or [`docker-internal`](https://github.com/ihmeuw-demographics/docker-internal)? If so follow directions in those repositories to rebuild and redeploy the images.
* [X] Do the changes require updates to other repositories which use this package? If yes, make the necessary updates in those repos, and consider integration tests for those repositories.

### CODE REVIEW

File: R/summarize_dt.R [hunk 1]
```diff
@@ -11,8 +11,10 @@
 #' @param summarize_cols \[`character()`\]\cr
 #'   The `id_cols` that should be collapsed and to calculate summary statistics
 #'   over.
-#' @param value_col \[`character(1)`\]\cr
-#'   Value column that summary statistics should be calculated for.
+#' @param value_cols \[`character()`\]\cr
+#'   Value columns that summary statistics should be calculated for. When more
```
Comment (chacalle) [inline R/summarize_dt.R:15]: Or should this be formatted long? The reason I am working on this was for updating `gen_summary_lt` https://github.com/ihmeuw-demographics/demCore/issues/67

Comment (chacalle) [reply R/summarize_dt.R:15]: Nevermind I guess this still allows users to use the function with long formatted data. And we just need to separately decide if `gen_summary_lt` should still return data long with the extra `life_table_parameter` column

Comment (krpaulson) [reply R/summarize_dt.R:15]: Do you need to edit the `@return` field too? To specify output will be wide on value column and summary function?

**Review by krpaulson — DISMISSED** [review]
It's fine by me to have this incorporated into the next build.

**Pull request merged**